### PR TITLE
DSC_CMSecurityRoles

### PR DESCRIPTION
**Pull request by jeffotterpohl**
#### Pull Request (PR) description
Initial creation of CMSecurityRoles.  The module will give the ability to import Security Roles or Append Security Roles with additional settings with specifying a xml file.  The module will also give the ability to remove Security roles if no users are currently assigned to the security role.  

#### Task list

- [X] Added an entry to the change log under the Unreleased section of the
      file CHANGELOG.md. Entry should say what was changed and how that
      affects users (if applicable), and reference the issue being resolved
      (if applicable).
- [X] Resource documentation added/updated in README.md.
- [X] Resource parameter descriptions added/updated in README.md, schema.mof
      and comment-based help.
- [X] Comment-based help added/updated.
- [X] Localization strings added/updated in all localization files as appropriate.
- [X] Examples appropriately added/updated.
- [X] Unit tests added/updated. See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [ ] Integration tests added/updated (where possible). See [DSC Community Testing Guidelines](https://dsccommunity.org/guidelines/testing-guidelines).
- [X] New/changed code adheres to [DSC Community Style Guidelines](https://dsccommunity.org/styleguidelines).

---
This change is [Reviewable](https://reviewable.io/reviews/dsccommunity/configmgrcbdsc/59)

### CODE REVIEW

File: source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1 [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+
+        if ($role.NumberOfAdmins -ge 1)
+        {
+            $users = Get-CMAdministrativeUser
+
+            foreach ($user in $users)
+            {
+                if ($user.RoleNames -Contains $SecurityRoleName)
+                {
+                    [array]$adminUsers += $user.logonName
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1:55]: $user.LogonName

File: source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1 [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+                    {
+                        Write-Warning -Message $script:localizedData.MissingGetConfig
+                        $applyXml = $true
+                    }
+                }
+            }
+
+            if ($applyXml)
+            {
+                Import-CMSecurityRole -XmlFileName $XmlPath -Overwrite $true
```
Comment (jeffotterpohl) [inline source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1:257]: OverWrite

File: source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1 [hunk 3]
```diff
@@ -0,0 +1,482 @@
...
+            }
+
+            $result = $false
+        }
+        else
+        {
+            if ($PSBoundParameters.ContainsKey('XmlPath') -and $OverWrite -eq $false -and $Append -eq $false)
+            {
+                Write-Warning -Message $script:localizedData.XmlFileNoOverwrite
+                $result = $false
```
Comment (NEllis280) [inline source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1:377]: Reminder: remove this line.

File: source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1 [hunk 4]
```diff
@@ -0,0 +1,482 @@
...
+    Import-ConfigMgrPowerShellModule -SiteCode $SiteCode
+    Set-Location -Path "$($SiteCode):\"
+    $state = Get-TargetResource -SiteCode $SiteCode -SecurityRoleName $SecurityRoleName
+    $result = $true
+
+    if ($Ensure -eq 'Present')
+    {
+        if ($state.Ensure -eq 'Absent')
+        {
+            if ([string]::IsNullOrEmpty($XMLPath))
```
Comment (NEllis280) [inline source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1:380]: $XmlPath

File: source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1 [hunk 5]
```diff
@@ -0,0 +1,482 @@
...
+                }
+                elseif ($PSBoundParameters.ContainsKey('XmlPath') -and ($OverWrite -eq $true -or $Append -eq $true))
+                {
+                    if ($state.Operation)
+                    {
+                        $convert = $state.Operation.TrimEnd(';').Split(';')
+                        $object = @{}
+                        foreach ($item in $convert)
+                        {
+                            $sitem = $item.Split('=')
```
Comment (NEllis280) [inline source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1:203]: $sItem?

File: source/Examples/Resources/CMSecurityRoles/CMSecurityRoles_Absent.ps1 [hunk 6]
```diff
@@ -0,0 +1,18 @@
+<#
+    .SYNOPSIS
+        A DSC configuration script to remove Security Roles to Configuration Manager.
```
Comment (NEllis280) [inline source/Examples/Resources/CMSecurityRoles/CMSecurityRoles_Absent.ps1:3]: *from Configuration Manager

File: source/Examples/Resources/CMSecurityRoles/CMSecurityRoles_Present.ps1 [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    .SYNOPSIS
+        A DSC configuration script to add Security Roles to Configuration Manager.
+#>
+Configuration Example
+{
+    Import-DscResource -ModuleName ConfigMgrCBDsc
+
+    Node localhost
+    {
+        CMSecurityRoles ExampleSettingsOverwrite
```
Comment (NEllis280) [inline source/Examples/Resources/CMSecurityRoles/CMSecurityRoles_Present.ps1:11]: OverWrite?

File: source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1 [hunk 8]
```diff
@@ -0,0 +1,482 @@
...
+                    Write-Verbose -Message ($script:localizedData.DescriptionSet -f $Description)
+                    Set-CMSecurityRole -Name $SecurityRoleName -Description $Description
+                }
+            }
+        }
+        elseif ($state.Ensure -eq 'Present')
+        {
+            if ($state.UsersAssigned)
+            {
+                Write-Warning -Message ($script:localizedData.RoleDeleteAdmin -f ($state.UsersAssigned | Out-String))
```
Comment (NEllis280) [inline source/DSCResources/DSC_CMSecurityRoles/DSC_CMSecurityRoles.psm1:290]: Did you mean to throw here?

**Pull request merged**